xtractmime/mimegroups.py

```diff
@@ -46,3 +52,8 @@ def is_javascript_mime_type(mime_type: bytes) -> bool:
 
 def is_json_mime_type(mime_type: bytes) -> bool:
     return mime_type.endswith(b"+json") or mime_type in (b"application/json", b"text/json")
+
+
+def is_text_mime_type(mime_type: bytes) -> bool:
+    text_types = tuple(text_type[3] for text_type in EXTRA_PATTERNS)
+    return mime_type in _APACHE_TYPES or mime_type in text_types
```

Comment:
Since this is not covered by the standard, I would not implement it. Later, if we have time, we can request that it be defined in the standard, and propose an implementation. It would probably be more complex, though, since most types are also valid text types (XML, HTML, JSON, JavaScript).

In the tests, we can still test them expecting an empty set as result. In Scrapy, if we need this, we can have a custom implementation there until it can be moved upstream to xtractmime following an extension of the standard.